modules/slider/Slider.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useEffect, useCallback } from "react";
+import { withStyles } from "@diana-ui/base";
+import { useWindowSize } from "@diana-ui/hooks";
+import {
+  ThemeStyleSheetFactory,
+  WithStylesProps,
+  StandardProps
+} from "@diana-ui/types";
+import { Label } from "@diana-ui/typography";
+
+export interface ISliderProps extends StandardProps<"input"> {
+  min: number;
+  max: number;
+  value?: number;
+  step: number;
+  disabled?: boolean;
+  className?: string;
+  inputClassName?: string;
+  onValueChange?: (value: number) => void;
+}
+
+const styleSheet: ThemeStyleSheetFactory = theme => ({
+  wrapper: {},
+  valueWrapper: {},
+  input: {
+    "&::-webkit-slider-thumb": {},
+    "@selectors": {
+      "&.disabled::-webkit-slider-thumb": {},
+      "&.disabled::-webkit-slider-runnable-track": {}
+    },
+    "&::-moz-range-thumb": {},
+    "&::-ms-thumb": {},
+    "&:focus": {},
+    "&::-ms-track": {},
+    "&::-webkit-slider-runnable-track": {}
+  },
+  value: {}
+});
+
+const Slider: React.FC<ISliderProps & WithStylesProps> = ({
+  cx,
+  styles,
+  min,
+  max,
+  value,
+  step,
+  onValueChange,
+  disabled = false,
+  className = "",
+  inputClassName = ""
+}) => {
+  const windowSize = useWindowSize();
+  const [size, setSize] = useState(9);
+  const [_value, setValue] = useState(value || 0);
+  const ref = React.createRef<HTMLInputElement>();
+
```

Comment:
you might need to force setValue(value) on an useEffect, otherwise if the value is changed on the parent, it won't get replaced, right?

example:

//everytime the parent sets a new value, it will have priority over its own state
useEffect(() => {
  if(value !== undefined) {
    setValue(value); 
  }
}, [value])